src/libpgmoneta/wal.c

```diff
@@ -420,6 +424,11 @@ pgmoneta_wal(int srv, char** argv)
       pgmoneta_consume_data_row_messages(ssl, socket, buffer, &end_of_timeline_response);
       timeline = atoi(pgmoneta_query_response_get_data(end_of_timeline_response, 0));
       xlogpos = pgmoneta_query_response_get_data(end_of_timeline_response, 1);
+
+      // Assign the current xlogpos to LSN
+      memset(config->servers[srv].current_wal_lsn, 0, MISC_LENGTH);
+      snprintf(config->servers[srv].current_wal_lsn, MISC_LENGTH, "%s", xlogpos);
```

Comment:
You are merely updating xlogpos at the beginning and the end of the streaming. You should also update that during the streaming, i.e. [inside the inner while loop](https://github.com/pgmoneta/pgmoneta/blob/1be259e66d61ef58cd828894e0e15ffc9723c42e/src/libpgmoneta/wal.c#L267). Here xlogptr can be converted to xlogpos - with some bit manipulation to extract high and low 32 bits.

Reply:
I know it's a lot of back and forth, but I think you are close.

Reply:
No problem. I'll look into it now. Thanks.